src/features/campaigns/hooks/useCreateCampaignActivity.ts

```diff
@@ -21,6 +23,18 @@ interface UseCreateCampaignActivityReturn {
   createSurvey: (surveyBody: ZetkinSurveyPostBody) => IFuture<ZetkinSurvey>;
 }
 
+async function fixNewCallAssignmentFilterConfig(
+  apiClient: IApiClient,
+  callAssignment: ZetkinCallAssignment
+): Promise<ZetkinCallAssignment> {
+  await apiClient.rpc(fixNewCallAssignmentFilterConfigRpc, {
+    callAssignmentId: callAssignment.id,
+    orgId: callAssignment.organization.id,
+    queryId: callAssignment.target.id,
+  });
+  return callAssignment;
```

Comment:
…here you return the call assignment the way it looked before, with the query not updated. So the store will not be updated to reflect the new assignment.

This is not a huge problem with the current UI (because trying to access the query will actually load the query from it's own API resource) but it's also an unnecessary mistake compared to if the RPC had returned the updated call assignment.